gui/ThemeEngine.cpp

```diff
@@ -988,63 +1009,83 @@ void ThemeEngine::drawLineSeparator(const Common::Rect &r) {
 	drawDD(kDDSeparator, r);
 }
 
-void ThemeEngine::drawCheckbox(const Common::Rect &r, const Common::String &str, bool checked, WidgetStateInfo state) {
+void ThemeEngine::drawCheckbox(const Common::Rect &r, const Common::String &str, bool checked, WidgetStateInfo state, bool rtl) {
 	if (!ready())
 		return;
 
 	Common::Rect r2 = r;
-	DrawData dd = kDDCheckboxDefault;
+	DrawData dd = rtl ? kDDCheckboxDefaultRTL : kDDCheckboxDefault;
 
 	if (checked)
-		dd = kDDCheckboxSelected;
+		dd = rtl ? kDDCheckboxSelectedRTL : kDDCheckboxSelected;
 
 	if (state == kStateDisabled)
-		dd = checked ? kDDCheckboxDisabledSelected : kDDCheckboxDisabled;
+		dd = checked ? rtl ? kDDCheckboxDisabledSelectedRTL : kDDCheckboxDisabledSelected : rtl ? kDDCheckboxDisabledRTL : kDDCheckboxDisabled;
 
 	const int checkBoxSize = MIN((int)r.height(), getFontHeight());
-
 	r2.bottom = r2.top + checkBoxSize;
-	r2.right = r2.left + checkBoxSize;
+
+	if (rtl) {
+		r2.left = r.right - checkBoxSize;
+		r2.right = r.right;
+	} else {
+		r2.right = r2.left + checkBoxSize;
+	}
 
 	drawDD(dd, r2);
 
-	r2.left = r2.right + checkBoxSize;
-	r2.right = r.right;
+	if (rtl) {
+		r2.left = r.left;
+		r2.right = r.right - (checkBoxSize * 2);
+	} else {
+		r2.left = r2.right + checkBoxSize;
+		r2.right = r.right;
+	}
 
 	if (r2.right > r2.left) {
-		drawDDText(getTextData(dd), getTextColor(dd), r2, str, true, false, _widgets[kDDCheckboxDefault]->_textAlignH,
+		drawDDText(getTextData(dd), getTextColor(dd), r2, str, true, false, _widgets[dd]->_textAlignH,
```

Comment:
Not sure if changing the default to using a particular dd's align is the best choice? Maybe I should choose from `kDDCheckboxDefault` or `kDDCheckboxDefaultRTL` to avoid some potential crashes?